apps/src/music/README.md

```diff
@@ -0,0 +1,59 @@
+# Music Lab
+
+<img width="1512" alt="Screenshot 2024-02-12 at 10 11 06 AM" src="https://github.com/code-dot-org/code-dot-org/assets/2205926/ab4c80eb-8824-457f-b0e2-cf330b5dbce0">
+
+## Overview
+
+This is the client-side code for **Music Lab**, as instantiated by a `music` level using [Lab2](../lab2/).
+
+**Music Lab** is a powerful lab for creating music using code.  It includes the following features:
+- Uses Google Blockly for editing user code.
+- The user code generates a timeline of audio events, which is rendered.
+- Uses WebAudio to play back the audio events.
+- The rendered timeline of audio events reflects code changes instantly.
+- Code can be changed while audio is playing.
+- Support for user triggered code.
+- Support for user-created functions on the main Blockly workspace.
+- A comprehensive programming model that supports playing sounds together or sequentially, in a fully composable fashion.
+- UI features that allow students to understand connections between code and output:
+  - audio events that highlight when the associated code block is clicked.
+  - code blocks that highlight when an associated audio event is clicked.
+- Custom blockly field implementations for specialized blocks:
+  - The `play` block opens a sound library explorer.
+  - The `play drums` block opens a drum pattern editor, and has a custom rendering of the drum pattern in the block itself.
+  - The `play notes` block opens a note editor, and has a custom rendering of the note pattern in the block itself.
+
+## Use of **Lab2**
+
+**Music Lab** is also the first lab to use the new [Lab2](/../lab2/) framework, and showcases some features and components unique to that framework, when used in a level progression:
+
+- A very simple layout manager that determines all panel sizes automatically, so that the student doesn't have to do any panel management.
+- A new instructions and feedback component.
+  - Vastly simpler than the legacy instructions, and avoids some of their [architectural challenges](https://github.com/code-dot-org/code-dot-org/pull/46325).
+  - Uses the new **Lab2** validation system to check current execution against a set of conditions to determine which feedback to display, and whether the level has been completed successfully.
+  - Text in instructions and feedback can be set up to display a callout arrow when clicked, allowing the text to refer to on-screen elements.  Details in https://github.com/code-dot-org/code-dot-org/pull/55755.
+- When switching between **Lab2** levels, such as `music` levels, no page reload is required, reducing server load and improving the user experience.
+
+## User programming model
+
+There was a multi-month effort to find the right programing model for this tool.  The following were the major models that we explored:
```

Comment:
Is there a dev doc which includes the models that were explored but not used and the reasons that decisions were made around UI that could be referenced here?

Reply:
I don't think we have a single centralized document that captures this.  Happy to capture some more of that as a follow-up.